pkg/operator/operator.go

```diff
@@ -382,7 +369,7 @@ func (optr *Operator) maoConfigFromInfrastructure() (*OperatorConfig, error) {
 		return nil, err
 	}
 
-	featureGate, err := getFeatureGate(optr.featureGateLister)
+	featureGate, err := optr.osClient.ConfigV1().FeatureGates().Get(context.Background(), "cluster", metav1.GetOptions{})
```

Comment:
I didn't understand why tbh, but the existing `getFeatureGate()` function didn't work in the unit test at leasy